operantanalysis/scripts/habit_training.py

```diff
@@ -0,0 +1,41 @@
+from operantanalysis import load_file, extract_info_from_file, reward_retrieval, lever_pressing
+import glob
+from tkinter import filedialog
+from tkinter import *  # noqa
+import pandas as pd
+import matplotlib
+matplotlib.use("TkAgg")
+from matplotlib import pyplot as plt  # noqa
+
+days = int(input("How many days would you like to analyze?"))
+column_list = ['Subject', 'Sex', 'Day', 'Training', 'Dippers', 'Dippers Retrieved', 'Retrieval Latency',
+               'Lever Presses']
+df = pd.DataFrame(columns=column_list)
```

Comment:
I see duplication (across scripts) - even more of a reason to modularize, e.g. extract that body within the loop into a function, while keeping the outside (all the imports, definitions, UI interactions) defined in a single function.
What you seems to have specific per script is what you analyze (body in the loop), and how you plot.  The rest is the same.
Could be done via pairs of functions given to the main "loopy" function, or even via classes.

Reply:
Added the function!

Reply:
Great to see the progress! ;) next time you might like to also either paste the checksum of the commit here or a PR so you know what function and where ;)  e.g. for this one `habit_training_function` added in d0ea94f09fa5d6fc354a18b5151fd6deb2f17a27

Reply:
Will do that in the future-thank you!